### PR TITLE
[flutter_tools] Make packages not required for local engine

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

A package_config file is not required for all commands, but some commands were made to require one when the --local-engine flag was provided. Revert this change and add a comment to the test to ensure it doesn't get accidentally updated again.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I also renamed "OrFail" since it doesn't always fail now

File: packages/flutter_tools/lib/src/dart/package_map.dart [hunk 1]
```diff
@@ -25,8 +25,12 @@ String _globalPackagesPath;
 
 /// Load the package configuration from [file] or throws a [ToolExit]
 /// if the operation would fail.
-Future<PackageConfig> loadPackageConfigOrFail(File file, {
+///
+/// If [nonFatal] is true, in the event of an error an empty package
+/// config is returned.
+Future<PackageConfig> loadPackageConfigWithLogging(File file, {
   @required Logger logger,
+  bool nonFatal = false,
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/dart/package_map.dart:33]: In some other places we call this kind of parameter `throwOnError`

Comment (zanderso) [reply packages/flutter_tools/lib/src/dart/package_map.dart:33]: Also, is this parameter used anywhere?

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart [hunk 2]
```diff
@@ -354,9 +354,10 @@ class FlutterCommandRunner extends CommandRunner<void> {
 
     if (engineSourcePath == null && globalResults['local-engine'] != null) {
       try {
-        final PackageConfig packageConfig = await loadPackageConfigOrFail(
+        final PackageConfig packageConfig = await loadPackageConfigWithLogging(
           globals.fs.file(globalPackagesPath),
           logger: globals.logger,
+          throwOnError: false,
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart:360]: Forgot to check in the actual change

**Review by zanderso — APPROVED** [review]

**Comment by jonahwilliams** [comment]
Landing change to kick tree

**Pull request merged**